glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -122,4 +127,91 @@ public class HttpPingUploader {
         dateFormatter.setLocalizedDateFormatFromTemplate("EEE, dd MMM yyyy HH:mm:ss z")
         return dateFormatter.string(from: date)
     }
+
+    /// This function deserializes and processes all of the serialized ping files.
+    ///
+    /// This function will ignore files that don't match the UUID regex and just delete them to
+    /// prevent files from polluting the ping storage directory.
+    func process() {
+        let pingDirectory = HttpPingUploader.getOrCreatePingDirectory()
+
+        do {
+            let storageDirectory = try FileManager.default.contentsOfDirectory(
+                atPath: pingDirectory.absoluteString
+            )
+
+            for file in storageDirectory {
+                if file.lastPathComponent.matches(Constants.filePattern) {
+                    NSLog("\(Constants.logTag) : Processing ping: \(file)")
+                    processFile(pingDirectory.appendingPathComponent(file)) { success, error in
+                        if !success {
+                            NSLog(
+                                "\(Constants.logTag) : Error processing ping file: \(file) - \(error.debugDescription)"
+                            )
+                        } else {
+                            do {
+                                try FileManager.default.removeItem(
+                                    atPath: pingDirectory.appendingPathComponent(file).absoluteString
+                                )
+                            } catch {
+                                print("\(Constants.logTag) : Error deleting ping file: \(file)")
```

Comment:
Oh a print. Guess we can all unify that into the same log call later.

Reply:
I think if we made a GleanLogger class, we could then have log levels like debug, error, etc. and could wrap that around whether we want to call `print()` or `NSLog()`

Reply:
I think we can address this as part of work for [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1580856).